lib/floe/workflow/runner/kubernetes.rb

```diff
@@ -66,7 +68,17 @@ def status!(runner_context)
         end
 
         def running?(runner_context)
-          %w[Pending Running].include?(runner_context.dig("container_state", "phase"))
+          phase              = runner_context.dig("container_state", "phase")
+          container_statuses = runner_context.dig("container_state", "containerStatuses") || []
+          container_reasons  = container_statuses.map { |status| status.dig("state", "waiting", "reason") }.compact
+
+          return false if RUNNING_PHASES.none?(phase)
+          # If an image failed to pull then the phase will be Pending but the
+          # container status will be waiting with a reason of ImagePullBackOff
+          # or ErrImagePull
+          return false if (container_reasons & FAILURE_REASONS).any?
```

Comment:
So, with more_core_extensions this would be 

```suggestion
          return false if container_reasons.include_any?(FAILURE_REASONS)
```